nilmtk/metergroup.py

```diff
@@ -234,8 +234,20 @@ def __getitem__(self, key):
                 if (set(group.identifier.meters) == key_meters):
                     return group
             raise KeyError(key)
-        # find MeterGroup from list of ElecMeterIDs
+        
         elif isinstance(key, list):
+            # Find specific appliance according to ElecMeter instance if appliance repeating more than once 
+            # If not specified then default
+            if(isinstance(key[0],str)):
+                match_meter=False
+                for meter in self.meters:
+                    if(meter.identifier.instance==key[2]):
+                        match_meter=True
+                        return meter
+                if match_meter!= True:
+                    return self[key[0]
```

Comment:
Missing `] ` at the end of the line. The code will crash as soon as this branch of the if statement is used.
Ideally, you would write a unit test to cover your new code.